src/frontend/hooks/blogs.js

```diff
@@ -0,0 +1,31 @@
+import { useEffect } from 'react';
+import api from '../api';
+import { useDispatch, useSelector } from 'react-redux';
+import * as blogInfoActions from '../state/blogs/actions';
+import * as blogInfoSelectors from '../state/blogs/selectors';
+
+const useBlogInfo = () => {
+  const dispatch = useDispatch();
+  const blogInfo = useSelector(blogInfoSelectors.allBlogInfo);
+  useEffect(() => {
+    (async () => {
+      try {
+        const blogInfoResponse = await api.blogs.getBlogs();
+
+        const newBlogInfo = blogInfoResponse.data;
+
+        dispatch(blogInfoActions.updateBlogInfo(newBlogInfo));
+        
+      } catch (err) {
+        console.error(err);
+      }
+    })();
```

Comment:
Just curious, do we need a self-calling function ?

I think useEffect would always get called when the component is rendered initially, so I don't think there is a need to wrap inside the `(async () => { .... })();`

I think `useEffect(async () => { ....` is enough. Please correct me if I'm wrong.